backends/backends.c

```diff
@@ -547,6 +589,18 @@ void *backends_main(void *ptr) {
             backend_set_kinesis_variables(&default_port,&backend_response_checker,&backend_request_formatter);
             break;
         }
+        case BACKEND_TYPE_MONGODB: {
+#if HAVE_MONGOC
+            if(!mongodb_init(mongodb_uri, mongodb_database, mongodb_collection))
+                do_mongodb = 1;
+            else
+                error("BACKEND: cannot initialize MongoDB backend");
+#else
+            error("BACKEND: MongoDB support isn't compiled");
+#endif // HAVE_MONGOC
+            backend_set_mongodb_variables(&default_port, &backend_response_checker, &backend_request_formatter);
```

Comment:
Considering what we have inside this function, in my opinion you need to call it inside the if of the line 594, because  we only need to set the Mongodb variables case we have it. 
The code that you wrote is not wrong, because it will not set the variables, but I do not think that the compiler will avoid to move these 3 variables to a register and call the function, so we will save some opcodes. No less important, after to see where this function is called, I think we can put the whole function "backend_set_mongodb_variables" inside an "#if HAVE_MONGOC", because I cannot see it been useful for other backends.

Reply:
Done